lib/SPIRV/SPIRVLowerConstExpr.cpp

```diff
@@ -149,25 +151,30 @@ void SPIRVLowerConstExpr::visit(Module *M) {
       for (unsigned OI = 0, OE = II->getNumOperands(); OI != OE; ++OI) {
         auto Op = II->getOperand(OI);
         auto *Vec = dyn_cast<ConstantVector>(Op);
-        if (Vec && std::all_of(Vec->op_begin(), Vec->op_end(),
-                               [](Value *V) { return isa<ConstantExpr>(V); })) {
+        if (Vec && std::all_of(Vec->op_begin(), Vec->op_end(), [](Value *V) {
+              return isa<ConstantExpr>(V) || isa<Function>(V);
+            })) {
           // Expand a vector of constexprs and construct it back with series of
           // insertelement instructions
-          std::list<Instruction *> ReplList;
-          std::transform(
-              Vec->op_begin(), Vec->op_end(), std::back_inserter(ReplList),
-              [LowerOp](Value *V) { return LowerOp(cast<ConstantExpr>(V)); });
+          std::list<Value *> OpList;
+          std::transform(Vec->op_begin(), Vec->op_end(),
+                         std::back_inserter(OpList),
+                         [LowerOp](Value *V) { return LowerOp(V); });
           Value *Repl = nullptr;
           unsigned Idx = 0;
-          for (auto V : ReplList)
+          std::list<Instruction *> ReplList;
+          for (auto V : OpList) {
+            if (auto *Inst = dyn_cast<Instruction>(V))
+              ReplList.push_back(Inst);
```

Comment:
```suggestion
            if (isa<Instruction>(V))
              ReplList.push_back(V);
```

No need to perform actual cast, since type of the list changed

Reply:
No, we can't push Value* to a list<Instruction*> w/o any explicit cast